MaxText/multihost_dataloading.py

```diff
@@ -105,3 +105,21 @@ def get_next_batch_sharded(
   input_gdas = jtu.tree_map_with_path(partial(_form_global_array, global_mesh = global_mesh), local_data)
 
   return input_gdas
+
+
+class MultiHostDataLoadIterator:
```

Comment:
@aireenmei be aware.

@ZhiyuLi-goog looks great!

Reply:
Hi @ZhiyuLi-goog , my grain changes were in this morning. I had some refactoring on tfds input pipeline as well. Looks like it may cause conflicts with your change. Let me know if you have questions / want to discuss any change